lib/puppet/functions/loadjson.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+# @summary
+#   Load a JSON file containing an array, string, or hash, and return the data
+#   in the corresponding native data type.
+#
+# @example Example Usage:
+#   $myhash = loadjson('/etc/puppet/data/myhash.json')
+#   $myhash = loadjson('https://example.local/my_hash.json')
+#   $myhash = loadjson('https://username:password@example.local/my_hash.json')
+#   $myhash = loadjson('no-file.json', {'default' => 'value'})
+Puppet::Functions.create_function(:loadjson) do
+  # @param path
+  #   A file path or a URL.
+  # @param default
+  #   The default value to be returned if the file was not found or could not
+  #   be parsed.
+  #
+  # @return
+  #   The data stored in the JSON file, the type depending on the type of data
+  #   that was stored.
+  dispatch :loadjson do
+    param 'String[1]', :path
+    optional_param 'Data', :default
+    return_type 'Data'
+  end
+
+  def loadjson(path, default = nil)
+    if path.start_with?('http://', 'https://')
+      require 'open-uri'
+      begin
+        content = URI.open(path) { |f| load_json_source(f) }
```

Comment:
What happened to all the basic authentication related code and tests?

Reply:
Good point. I assumed it would just use the credentials from the URI, but it doesn't.